libdnf5/base/base.cpp

```diff
@@ -183,7 +195,17 @@ void Base::setup() {
     auto & installroot = config.get_installroot_option();
     installroot.lock("Locked by Base::setup()");
 
-    get_vars()->load(vars_installroot, config.get_varsdir_option().get_value());
+    // Initialize rpm before calling vars->load(), because it also detects
+    // default arch and releasever values, and the detection relies on rpm
+    // being initialized.
+    auto vars = get_vars();
+    const char * arch = nullptr;
+    if (vars->contains("arch")) {
+        arch = vars->get_value("arch").c_str();
+    }
+    init_lib_rpm(arch);
```

Comment:
I have additional  comment. If I am not mistaken, we pass rpm an architecture, but in this time we can still use a different architecture, because architecture might be modify or set by `vars->load(vars_installroot, config.get_varsdir_option().get_value());`. It means that we might use for POOL a different architecture that we provide to RPM. @pmatilai If we will use the same architecture in libsolv pool and we will initialize rpm config `rpmReadConfigFiles(nullptr, arch)` with the same architecture, could we stop use `RPMPROB_FILTER_IGNOREARCH` for transaction where architecture used for transaction calculation differs from the system architecture.

Reply:
Gosh, that's right! Thanks for catching it. So I'll move the code back to vars.cpp and just make sure that the code is called unconditionally.

Reply:
@j-mracek the issue with architecture provided by vars file should be resolved. Now the rpm is initialized during Vars::detect_vars() call.